ydb/_topic_writer/topic_writer_asyncio.py

```diff
@@ -580,32 +570,52 @@ class WriterAsyncIOStream:
     supported_codecs: Optional[List[PublicCodec]]
 
     _stream: IGrpcWrapperAsyncIO
-    _token_getter: TokenGetterFuncType
     _requests: asyncio.Queue
     _responses: AsyncIterator
 
+    _update_token_interval: int
+    _update_token_task: asyncio.Task
+    _update_token_event: asyncio.Event
+    _get_token_function: Callable[[], str]
+
     def __init__(
-        self,
-        token_getter: TokenGetterFuncType,
+        self, update_token_interval: int, get_token_function: Callable[[], str]
     ):
-        self._token_getter = token_getter
+        self._update_token_interval = update_token_interval
+        self._get_token_function = get_token_function
+
+        self._update_token_event = asyncio.Event()
+        self._update_token_event.set()
+
+        self._update_token_task = asyncio.create_task(
+            self._update_token_loop(), name="update_token_loop"
+        )
+
+    async def _update_token_loop(self):
```

Comment:
_update_token_loop must start from _start (and placed under _start) - after received init_response, or it will race condition between init-request/response and send update token.

If update token interval is small - first event may fired before init_request.

For example in tests, where interval may set to 0 (for update token in every event loop).

Reply:
right, thank you